src/Core/Framework/DataAbstractionLayer/DefinitionValidator.php

```diff
@@ -956,6 +958,57 @@ private function validateColumn(EntityDefinition $definition): array
         return [$definition->getClass() => $notices];
     }
 
+    /**
+     * Validates that PrimaryKey flags in entity definition match the database PRIMARY KEY constraint
+     *
+     * @return array<class-string<EntityDefinition>, list<string>>
+     */
+    private function validatePrimaryKeyConsistency(EntityDefinition $definition): array
```

Comment:
would be nice, if we can wait until I finished the DBAL update, because this will also have effects on this method :grimacing:

Reply:
Fine for me. Just tell me when you're ready. Thanks.